internal/acs/runner.go

```diff
@@ -0,0 +1,250 @@
+				}
+			}
+		}
+
+	}
+	return metricMap
+}
+
+func processClusterCVEs(cluster acsclient.Cluster, client acsclient.Client) (map[string]cveSummaryData, error) {
+	query := acsclient.Query{
+		"Cluster": cluster.Name,
+	}
+	exportDeployments, err := client.GetExportDeployments(query)
+	if err != nil {
+		return nil, fmt.Errorf("errors getting export deployments:: %v", err)
+	}
+	exportImages, err := client.GetExportImages(query)
+	if err != nil {
+		return nil, fmt.Errorf("errors getting export images:: %v", err)
+	}
+	exportDeploymentsMap := getexportDeploymentsMap(exportDeployments)
+	exportImagesMap := getExportImagesMap(exportImages)
+	return processNamespaces(query, client, exportDeploymentsMap, exportImagesMap)
+}
+
+func getExportImagesMap(exportImages []acsclient.ExportImage) map[string]acsclient.ExportImage {
+	imagesMap := make(map[string]acsclient.ExportImage)
+	for _, exportImage := range exportImages {
+		imagesMap[exportImage.ID] = exportImage
+	}
+	return imagesMap
+}
+
+func getexportDeploymentsMap(exportDeployments []acsclient.ExportDeployment) map[string][]acsclient.ExportDeployment {
+	exportDeploymentsMap := make(map[string][]acsclient.ExportDeployment)
+	for _, exportDeployment := range exportDeployments {
+		deployments, exists := exportDeploymentsMap[exportDeployment.Namespace]
+		if exists {
+			deployments = append(deployments, exportDeployment)
+			exportDeploymentsMap[exportDeployment.Namespace] = deployments
+		} else {
+			exportDeploymentsMap[exportDeployment.Namespace] = []acsclient.ExportDeployment{exportDeployment}
+		}
+	}
+	return exportDeploymentsMap
+}
+func processNamespaces(query acsclient.Query, client acsclient.Client, exportDeploymentsMap map[string][]acsclient.ExportDeployment, exportImagesMap map[string]acsclient.ExportImage) (map[string]cveSummaryData, error) {
+	namespaces, err := client.GetNamespaces(query)
+	if err != nil {
+		return nil, fmt.Errorf("errors getting namespaces:: %v", err)
+	}
+	cveMap := map[string]cveSummaryData{}
+	for _, namespace := range namespaces {
+		query["Namespace"] = namespace.Metadata.Name
+		deployments := exportDeploymentsMap[namespace.Metadata.Name]
+		if len(deployments) > 0 {
+			summarizeCVEData(query, cveMap, deployments, exportImagesMap)
+		}
+
+	}
+	return cveMap, nil
+}
+
+func summarizeCVEData(query acsclient.Query, cveMap map[string]cveSummaryData, exportDeployments []acsclient.ExportDeployment, exportImagesMap map[string]acsclient.ExportImage) {
+	for _, deployment := range exportDeployments {
+		populateCveMap(query, deployment, cveMap, exportImagesMap)
+	}
+}
+
+func populateCveMap(query acsclient.Query, deployment acsclient.ExportDeployment, cveMap map[string]cveSummaryData, exportImagesMap map[string]acsclient.ExportImage) {
+	for _, container := range deployment.Containers {
+		index := fmt.Sprintf("%s|%s|%s", query["Cluster"], query["Namespace"], container.Image.ID)
+		data, exists := cveMap[index]
+		if exists {
+			data.instances[deployment.ID] += 1
```

Comment:
why do we need to use `deployment.ID` as key? can `instances` just be an int counter?

Reply:
this `+= 1` result it not used, it's actually just a set to keep deployment ID, we can use bool or empty struct as value, only need action on not exists path.